webSockets/server/server.go

```diff
@@ -0,0 +1,209 @@
+}
+
+func (s *server) defaultConnectionHandler(conn webSockets.WSConClient) {
+	_ = s.sender.AddConnection(conn)
+}
+
+func (s *server) initializeServer(wsURL string, wsPath string) {
+	router := mux.NewRouter()
+	httpServer := &http.Server{
+		Addr:    wsURL,
+		Handler: router,
+	}
+
+	upgrader := websocket.Upgrader{
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+	}
+
+	s.log.Info("wsServer.initializeServer(): initializing web-sockets server", "url", wsURL, "path", wsPath)
+
+	addClientFunc := func(writer http.ResponseWriter, r *http.Request) {
+		s.log.Info("new connection", "route", wsPath, "remote address", r.RemoteAddr)
+
+		upgrader.CheckOrigin = func(r *http.Request) bool { return true }
+
+		ws, errUpgrade := upgrader.Upgrade(writer, r, nil)
+		if errUpgrade != nil {
+			s.log.Warn("could not update websocket connection", "remote address", r.RemoteAddr, "error", errUpgrade)
+			return
+		}
+		client := connection.NewWSConnClientWithConn(ws)
+		s.connectionHandler(client)
+	}
+
+	routeSendData := router.HandleFunc(wsPath, addClientFunc)
+	if routeSendData.GetError() != nil {
+		s.log.Error("sender router failed to handle send data",
+			"route", routeSendData.GetName(),
+			"error", routeSendData.GetError())
+	}
+
+	s.httpServer = httpServer
+}
+
+// Send will send the provided payload from args
+func (s *server) Send(args data.WsSendArgs) error {
+	return s.sender.Send(args)
+}
+
+// Start will start the websockets server
+func (s *server) Start() {
+	err := s.httpServer.ListenAndServe()
+	shouldLogError := err != nil && !strings.Contains(err.Error(), data.ErrServerIsClosed.Error())
+	if shouldLogError {
+		s.log.Error("could not initialize webserver", "error", err)
+		return
+	}
+
+	s.log.Info("server was closed")
+}
+
+// SetPayloadHandler will set the provided payload handler
+func (s *server) SetPayloadHandler(handler webSockets.PayloadHandler) error {
+	s.payloadHandler = handler
+	return nil
+}
+
+// Listen will switch the server in listen mode and the server will start to listen from messages from the new connections
+func (s *server) Listen() {
+	// TODO refactor this method
+	s.connectionHandler = func(connection webSockets.WSConClient) {
+		webSocketsReceiver, err := receiver.NewReceiver(receiver.ArgsReceiver{
+			PayloadConverter:   s.payloadConverter,
+			Log:                s.log,
+			RetryDurationInSec: int(s.retryDuration.Seconds()),
+			BlockingAckOnError: s.blockingAckOnError,
+		})
+		if err != nil {
+			s.log.Warn("s.connectionHandler cannot create receiver", "error", err)
+		}
+		err = webSocketsReceiver.SetPayloadHandler(s.payloadHandler)
+		if err != nil {
+			s.log.Warn("s.SetPayloadHandler cannot set payload handler", "error", err)
+		}
+
+		go func() {
+			s.receivers.AddReceiver(connection.GetID(), webSocketsReceiver)
+			// this method is blocking
+			_ = webSocketsReceiver.Listen(connection)
+			// if method listen will end, the client was disconnected should remove the listener from the list
```

Comment:
// if method listen will end, the client was disconnected and we should remove the listener from the list